lib/repl.js

```diff
@@ -496,6 +506,44 @@ function REPLServer(prompt,
     self.displayPrompt(true);
   });
 
+  // Wrap readline tty to enable editor mode
+  const ttyWrite = self._ttyWrite.bind(self);
+  self._ttyWrite = (d, key) => {
+    if (!self.editorMode || !self.terminal) {
+      ttyWrite(d, key);
+      return;
+    }
+
+    // editor mode
+    if (key.ctrl && !key.shift) {
+      switch (key.name) {
+        case 'd': // End editor mode
+          self.turnOffEditorMode();
+          sawCtrlD = true;
+          ttyWrite(d, { name: 'return' });
+          break;
+        case 'n': // Override next history item
+        case 'p': // Override previous history item
+          break;
+        case 'z':
+          if (process.platform !== 'win32') {
+            self.turnOffEditorMode();
```

Comment:
I turned off the editor mode on suspending (`Ctrl + Z`) the process.  In windows, we are not taking any [action](https://github.com/nodejs/node/blob/master/lib/readline.js#L767).

I am updating the PR to resume `.editor` mode on `SIGCONT`.